components/dashboards-web-component/src/designer/components/PagesSidePane.jsx

```diff
@@ -170,6 +179,37 @@ class PagesSidePane extends Component {
         });
     }
 
+    /**
+     * Update dashboard page visibility.
+     *
+     * @param {string} pageId Page ID
+     * @param {boolean} hidden Is hidden
+     * @returns {Promise}
+     */
+    handlePageVisibility(pageId, hidden) {
+        return new Promise((resolve, reject) => {
+            const page = this.getPageById(pageId);
+            if (!page) {
+                this.setValidationErrorMessage(`Cannot find page for ID '${pageId}'!`);
+                reject();
+            }
+            page.hidden = hidden;
+            this.props.updateDashboard()
+                .then(() => resolve())
+                .catch(() => {
+                    delete page.title;
```

Comment:
It should be `delete page.hidden`. Thanks for finding that out!

Reply:
Why `delete page.hidden`?

IMO, instead of `delete` a property we should set an appropriate value to it. In this case, lets always set `page.hidden` to `true` or `false`. This will automatically migrate the dashboard JSON to new structure.